### DISCUSSION TITLE
[Dev Server] Migrating the deprecated `http2` and `https` options to the `server` option

### DISCUSSION THREAD

**Discussion by egonolieux** [webpack/webpack#14949] [Q&A]
I migrated the following deprecated options of webpack dev server (see https://webpack.js.org/configuration/dev-server/#devserverhttp2):

```js
// Deprecated options.

//options.http2 = true,
//options.https = {
//    key: 'infrastructure/ssl/private/dev.key',
//    cert: 'infrastructure/ssl/certs/dev.crt',
//},

// New options.

options.server = {
    type: 'https',
    options: {
        key: 'infrastructure/ssl/private/dev.key',
        cert: 'infrastructure/ssl/certs/dev.crt',
    },
}
```

However, when starting the dev server, I'm getting mixed content errors because somehow the dev server is using plain http? Using the deprecated options works fine though.

```txt
Mixed Content: The page at 'https://localhost/dashboard/intl' was loaded over HTTPS, but requested an insecure element 'http://localhost:8080/build/images/logo.6676f389.png'. This request was automatically upgraded to HTTPS, For more information see https://blog.chromium.org/2019/10/no-more-mixed-messages-about-https.html
```

 Am I missing something here?

**Comment by snitin315** [comment]
If you previously had `http2` enabled via `http2: true`,  your new config should be -

```js
options.server = {
    type: 'spdy',
    options: {
        key: 'infrastructure/ssl/private/dev.key',
        cert: 'infrastructure/ssl/certs/dev.crt',
    },
}
```

**Reply by egonolieux** [reply]
Yes, but I do not intend on using `spdy` specifically, the issue is the same for both `https` and `spdy`.

**Reply by alexander-akait** [reply]
Can you provide example, because we don't change logic, only refactor options

**Reply by egonolieux** [reply]
> I digged into the issue more deeply, and this seems to be an issue with Webpack Encore specifically (Symfony): https://github.com/symfony/webpack-encore/blob/main/lib/config-generator.js#L67. In other words, this has nothing to do with Webpack itself.
